05.wc/wc.rb

```diff
@@ -0,0 +1,78 @@
+# ! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+SEPARATOR_SIZE = 1
+STDIN_FILENAME = '-'
+TOTAL = '合計'
+
+def main
+  options = parse_options(ARGV)
+  rows = []
+  ARGF.each(nil) do |input_text|
+    counts = {}
+
+    counts[:filename] = ARGF.filename
+    counts[:line] = input_text.lines.count if options[:l]
+    counts[:word] = input_text.split(/\s+/).size if options[:w]
+    counts[:byte] = input_text.size if options[:c]
+    rows << counts
+  end
+  rows << calculate_total_rows(rows) if rows.size > 1
+  max_widths = calculate_max_widths(rows)
+  show_rows(rows, max_widths)
+end
+
+def calculate_total_rows(rows)
+  total = rows.inject({}) do |result, count|
+    result.merge(count) do |key, current_val, adding_value|
+      current_val + adding_value unless key == :filename
+    end
+  end
+  total[:filename] = TOTAL
+  total
+end
+
+def show_rows(rows, max_column_widths)
+  rows.each do |row|
```

Comment:
```suggestion
  rows.each do |counts|
```

rowsの中身はcountsと呼ぶ、と一貫性を持たせると良いかもと思いました